openmc/kinetics/solver.py

```diff
@@ -0,0 +1,886 @@
+    def materials_file(self):
+        return self._materials_file
+
+    @property
+    def transient(self):
+        return self._transient
+
+    @property
+    def mgxs_lib_file(self):
+        return self._mgxs_lib_file
+
+    @property
+    def clock(self):
+        return self._clock
+
+    @property
+    def one_group(self):
+        return self._one_group
+
+    @property
+    def energy_groups(self):
+        return self._energy_groups
+
+    @property
+    def fine_groups(self):
+        return self._fine_groups
+
+    @property
+    def tally_groups(self):
+        return self._tally_groups
+
+    @property
+    def initial_power(self):
+        return self._initial_power
+
+    @property
+    def k_crit(self):
+        return self._k_crit
+
+    @property
+    def mpi_args(self):
+        return self._mpi_args
+
+    @property
+    def threads(self):
+        return self._threads
+
+    @property
+    def chi_delayed_by_delayed_group(self):
+        return self._chi_delayed_by_delayed_group
+
+    @property
+    def use_agd(self):
+        return self._use_agd
+
+    @property
+    def use_pcmfd(self):
+        return self._use_pcmfd
+
+    @property
+    def chi_delayed_by_mesh(self):
+        return self._chi_delayed_by_mesh
+
+    @property
+    def num_delayed_groups(self):
+        return self._num_delayed_groups
+
+    @property
+    def states(self):
+        return self._states
+
+    @property
+    def use_pregenerated_sps(self):
+        return self._use_pregenerated_sps
+
+    @property
+    def constant_seed(self):
+        return self._constant_seed
+
+    @property
+    def seed(self):
+        return self._seed
+
+    @property
+    def core_volume(self):
+        return self._core_volume
+
+    @property
+    def log_file_name(self):
+        return self._log_file_name
+
+    @property
+    def run_on_cluster(self):
+        return self._run_on_cluster
+
+    @property
+    def job_file(self):
+        return self._job_file
+
+    @property
+    def multi_group(self):
+        return self._multi_group
+
+    @property
+    def inner_tolerance(self):
+        return self._inner_tolerance
+
+    @property
+    def outer_tolerance(self):
+        return self._outer_tolerance
+
+    @property
+    def method(self):
+        return self._method
+
+    @property
+    def min_outer_iters(self):
+        return self._min_outer_iters
+
+    @directory.setter
+    def directory(self, directory):
+        self._directory = directory
+
+    @amplitude_mesh.setter
+    def amplitude_mesh(self, mesh):
+
+        self._amplitude_mesh = openmc.RegularMesh()
+        self._amplitude_mesh.type = mesh.type
+        self._amplitude_mesh.dimension = mesh.dimension
+        self._amplitude_mesh.lower_left = mesh.lower_left
+
+        unity_mesh = openmc.RegularMesh()
+        unity_mesh.type = mesh.type
+        unity_mesh.dimension = [1,1,1]
+        unity_mesh.lower_left  = mesh.lower_left
+        if mesh.width is not None:
```

Comment:
I believe you can also do simply `if mesh.width` here if you want to make it more concise